molior/api/sourcerepository.py

```diff
@@ -28,6 +31,18 @@ def get_last_build(db, projectversion, repository):
     return last_build
 
 
+def get_latest_release(db, projectversion, repository):
+    candidates = db.query(Build).filter(Build.sourcerepository_id == repository.id,
+                                        Build.projectversion_id == projectversion.id,
+                                        Build.buildtype == "deb", Build.buildstate == "successful").all()
+    iterator = iter(candidates)
+    m = next(iterator, None)
+    for item in iterator:
+        if apt_pkg.version_compare(item.version, m.version) > 0:
+            m = item
```

Comment:
This whole logic of finding a max could be done more clearly like this:
```
m = max(candidates, key=lambda item: functools.cmp_to_key(apt_pkg.version_compare)(item.version))
```
i.e. by finding the max of the sequence of candidates by using as comparator the function `apt_pkg.version_compare`.

Reply:
m = max(candidates, key=lambda item: functools.cmp_to_key(apt_pkg.version_compare)(item.version), default=None)